uniswap_universal_router_decoder/_abi_builder.py

```diff
@@ -340,6 +341,16 @@ def _build_permit2_transfer_from() -> FunctionABI:
         builder = FunctionABIBuilder(RouterFunction.PERMIT2_TRANSFER_FROM.name)
         return builder.add_address("token").add_address("recipient").add_uint256("amount").build()
 
+    @staticmethod
+    def _build_permit2_transfer_from_batch() -> FunctionABI:
+        builder = FunctionABIBuilder(RouterFunction.PERMIT2_TRANSFER_FROM_BATCH.name)
+        return builder.add_struct_array(_ABIBuilder._allowance_transfer_details_struct_array_builder()).build()
+
+    @staticmethod
+    def _allowance_transfer_details_struct_array_builder() -> FunctionABIBuilder:
+        builder = FunctionABIBuilder.create_struct_array("AllowanceTransferDetails")
+        return builder.add_address("from").add_address("to").add_uint256("amount").add_address("token")
```

Comment:
as per the Permit2 [contract](https://github.com/Uniswap/permit2/blob/cc56ad0f3439c502c246fc5cfcc3db92bb8b7219/src/interfaces/IAllowanceTransfer.sol#L103), the amount is an `uint160`